ethers-contract/src/contract.rs

```diff
@@ -145,37 +145,73 @@ use ethers_core::types::TransactionRequest;
 ///
 /// _Disclaimer: these above docs have been adapted from the corresponding [ethers.js page](https://docs.ethers.io/ethers.js/html/api-contract.html)_
 ///
+/// # Usage Note
+///
+/// `ContractInternal` accepts any client that implements `B: Borrow<M>` where
+/// `M :Middleware`. Previous `Contract` versions used only arcs, and relied
+/// heavily on [`Arc`]. Due to constraints on the [`ContractCallInternal`] type,
+/// calling contracts requires a `B: Borrow<M> + Clone`. This is fine for most
+/// middlware. However, when `B` is an owned middleware that is not Clone, we
+/// cannot issue contract calls. Some notable exceptions:
+///
+/// - `NonceManagerMiddleware`
+/// - `SignerMiddleware` (when using a non-Clone Signer)
+///
+/// When using non-Clone middlewares, instead of instantiating a contract that
+/// OWNS the middlware, pass the contract a REFERENCE to the middleware. This
+/// will fix the trait bounds issue (as `&M` is always `Clone`).
+///
+/// We expect to fix this fully in a future version
+///
 /// [`abigen`]: macro.abigen.html
 /// [`Abigen` builder]: struct.Abigen.html
 /// [`event`]: method@crate::Contract::event
 /// [`method`]: method@crate::Contract::method
 #[derive(Debug)]
-pub struct Contract<M> {
+pub struct ContractInternal<B, M> {
     address: Address,
     base_contract: BaseContract,
-    client: Arc<M>,
+    client: B,
+    _m: PhantomData<M>,
 }
 
-impl<M> std::ops::Deref for Contract<M> {
+/// `Contract` is a [`ContractInternal`] object with an `Arc` middleware.
+/// This type alias exists to preserver backwards compatibility with
+/// non-abstract Contracts.
+///
+/// For full usage docs, see [`ContractInternal`]
+pub type Contract<M> = ContractInternal<std::sync::Arc<M>, M>;
```

Comment:
should we perhaps mark the aliases as deprecated?

Reply:
adding a deprecated tag results in close to 200 warnings in the ethers codebase. To avoid noise in the PR, I am adding in a commented-out deprecated attribute. I'll uncomment it and change all of our internal usage as a last step.

Reply:
update:

see top comment for future plans wrt this. Aliases will not be deprecated in this PR. We will deprecate after we have a straightforward and beginner-friendly migration plan